Source/HelixToolkit.Wpf.SharpDX/Model/Elements3D/PointGeometryModel3D.cs

```diff
@@ -314,35 +314,33 @@ public override void Dispose()
         /// <summary>
         /// Creates a <see cref="T:PointsVertex[]"/>.
         /// </summary>
-        private Geometry3D.PointsVertex[] CreatePointVertexArray()
+        private PointsVertex[] CreatePointVertexArray()
         {
-            var positions = this.Geometry.Positions.Array;
-            var vertexCount = this.Geometry.Positions.Count;
-            var color = this.Color;
-            var result = new Geometry3D.PointsVertex[vertexCount];
+            var positions = Geometry.Positions.Array;
+            var vertexCount = Geometry.Positions.Count;
+            var color = Color;
+            var result = new PointsVertex[vertexCount];
+            var colors = Geometry.Colors;
 
-            if (this.Geometry.Colors != null && this.Geometry.Colors.Any())
+            for (var i = 0; i < vertexCount; i++)
             {
-                var colors = this.Geometry.Colors;
-                for (var i = 0; i < vertexCount; i++)
+                Color4 finalColor;
+                if (colors != null && colors.Any())
                 {
-                    result[i] = new Geometry3D.PointsVertex
-                    {
-                        Position = new Vector4(positions[i], 1f),
-                        Color = color * colors[i],
-                    };
+                    
+                    finalColor = color*colors[i];
                 }
-            }
-            else
-            {
-                for (var i = 0; i < vertexCount; i++)
+                else
                 {
-                    result[i] = new Geometry3D.PointsVertex
-                    {
-                        Position = new Vector4(positions[i], 1f),
-                        Color = color,
-                    };
+                    finalColor = color;
                 }
+
+                result[i] = new PointsVertex
+                {
+                    Position = new Vector4(positions[i], 1f),
+                    Color = finalColor,
+                    Parameters = new Vector4(IsSelected?1:0,0,0,0)

```

Comment:
The `PointsVertex.Parameters` field now stores a vector for conveying extra data to the shader. Currently, we only use one component of this vector to store the `IsSelected` value.